state/store.go

```diff
@@ -651,14 +651,22 @@ func (store dbStore) LoadFinalizeBlockResponse(height int64) (*abci.FinalizeBloc
 
 	resp := new(abci.FinalizeBlockResponse)
 	err = resp.Unmarshal(buf)
+	// if an error is returned, then it needs to be handled here
 	if err != nil {
+		return nil, ErrFinalizeBlockResponseUnmarshalError
```

Comment:
I've refactored the errors a bit. I didn't implement a `Wrap()` because it would surface the internal storage error to the API response call which is usually not a best practice. I think it's better to log the error from the store and then just return additional context (height) as the other errors are doing